object_intern.go

```diff
@@ -404,23 +505,26 @@ func (oi *ObjectIntern) DeleteByByte(obj []byte) (bool, error) {
 //
 // false, error - the object was not found in the object store or could not be deleted
 func (oi *ObjectIntern) DeleteByString(obj string) (bool, error) {
-	var addr gos.ObjAddr
-	var ok bool
 
 	if oi.conf.Compression != None {
-		obj = string(oi.compress([]byte(obj)))
+		oi.RLock()
+		// try to find the compressed object in the index
+		addr, ok := oi.objIndex[string(oi.compress([]byte(obj)))]
```

Comment:
to not have to have duplications of the same logic, once with compression and once without, I'd add a method like `deleteByString()` which requires the `obj` to not be compressed, then `DeleteByString()` can call that either directly or after first compressing `obj`. 
the same probably applies to other methods as well

Reply:
Yeah, it depends on complexity and speed for me. I'll know more after running the benchmarks for time with various changes. Once I get stable results we can decide.

Reply:
I will possibly address this in another PR